docs/_includes/Statements.md

```diff
@@ -491,20 +544,20 @@ correct guess by Dafny.
 
 A `decreases` annotation, as its name suggests, gives Dafny an expression
 that decreases with every loop iteration or recursive call. There are two
-conditions that Dafny needs to verify when using a decreases expression:
+conditions that Dafny needs to verify when using a `decreases` expression:
 
 * that the expression actually gets smaller, and
 * that it is bounded.
```

Comment:
This is true. Using more mathematical language, these two bullets can be summarized by saying the value with each successive iteration must go strictly down in a well-founded order.

Upon reflection, I think it the reference manual should mention well-founded orders. Dafny builds in a well-founded order for every type. It cannot be changed, but a program can choose how to maps values into the built-in well-founded orders. This is done by giving an expression in a `decreases` clause.

The well-founded orders of all the types combined into one "big" well-founded order over all possible values of any type. This "big" well-founded order sometimes relates values from different types; for example, a `Color` datatype value `Blue` lies below the `List<Color>` datatype value `Cons(Blue, Nil)`.

Dafny allows a termination metric to be a listing of expressions (of any types), each of which is ordered by the "big" well-founded order. The listing of values is ordered as lexicographically, but where a listing is consider smaller than all its proper prefixes. Since any program has a bound on the length of these listings, the ordering on listings is well-founded.

Reply:
Added the mathematical language and put in a place-holder section on Well-founded orders